From 2793289fe3e92da3f5ea94af9bdd7a904b70e4aa Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Fri, 5 Oct 2018 19:07:56 +0300
Subject: [PATCH 4840/5242] MLK-18497-9: ASoC: fsl: dsp: Export
 xf_client_alloc

commit  454123b0c96b27c2b25114a9f95431475ed01939 from
https://source.codeaurora.org/external/imx/linux-imx.git

While at it add include guard to dsp.h

Reviewed-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_dsp.c |    9 +++++++--
 sound/soc/fsl/fsl_dsp.h |    7 ++++++-
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/fsl_dsp.c b/sound/soc/fsl/fsl_dsp.c
index e8a77eb..b1fc3d1 100644
--- a/sound/soc/fsl/fsl_dsp.c
+++ b/sound/soc/fsl/fsl_dsp.c
@@ -62,11 +62,16 @@
 #include <soc/imx8/sc/svc/irq/api.h>
 #include <soc/imx8/sc/ipc.h>
 #include <soc/imx8/sc/sci.h>
-#include "fsl_dsp.h"
 
+#include <sound/pcm.h>
+#include <sound/soc.h>
+
+#include "fsl_dsp.h"
+#include "fsl_dsp_pool.h"
+#include "fsl_dsp_xaf_api.h"
 
 /* ...allocate new client */
-static inline struct xf_client *xf_client_alloc(struct fsl_dsp *dsp_priv)
+struct xf_client *xf_client_alloc(struct fsl_dsp *dsp_priv)
 {
 	struct xf_client *client;
 	u32             id;
diff --git a/sound/soc/fsl/fsl_dsp.h b/sound/soc/fsl/fsl_dsp.h
index 99f92b3..c6aa1eb 100644
--- a/sound/soc/fsl/fsl_dsp.h
+++ b/sound/soc/fsl/fsl_dsp.h
@@ -5,8 +5,10 @@
  *
  */
 
-#include <soc/imx8/sc/ipc.h>
+#ifndef FSL_DSP_H
+#define FSL_DSP_H
 #include <uapi/linux/mxc_dsp.h>
+#include <soc/imx8/sc/ipc.h>
 #include "fsl_dsp_proxy.h"
 
 
@@ -119,3 +121,6 @@ struct fsl_dsp {
 void *memcpy_dsp(void *dest, const void *src, size_t count);
 void *memset_dsp(void *dest, int c, size_t count);
 struct xf_client *xf_client_lookup(struct fsl_dsp *dsp_priv, u32 id);
+struct xf_client *xf_client_alloc(struct fsl_dsp *dsp_priv);
+
+#endif
-- 
1.7.9.5

